ui-particles-angular/projects/ui-particles-angular/src/lib/gio-license/gio-license.directive.ts

```diff
@@ -37,13 +41,20 @@ export class GioLicenseDirective implements OnInit, OnDestroy {
   constructor(private readonly licenseService: GioLicenseService, private readonly matDialog: MatDialog, private elRef: ElementRef) {}
 
   public ngOnInit(): void {
-    this.isMissingFeature$()
+    const gioLicense = this.gioLicense;
+
+    // If the license is not defined, we do not need to check anything
+    if (isNil(gioLicense)) {
+      return;
+    }
+
+    this.isMissingFeature$(gioLicense)
       .pipe(
         tap(() => {
           this.elRef.nativeElement.removeEventListener('click', this.onClick, true);
         }),
         filter(notAllowed => this.gioLicense != null && notAllowed),
```

Comment:
Is the `this.gioLicense != null` still necessary?

Reply:
no you right ;)

Reply:
The idea behind the filter is to make sure you unbind the click every time.
It might be better to move the isNull check to the isMissingFeature$ and return false.

Reply:
OK, I see, it seems OK like that are we are inside ngOnInit. If we move on to ngOnChanges, we may have to review the observable, but that's to be seen later.